Branch-SDK/src/main/java/io/branch/referral/BranchActivityLifecycleObserver.java

```diff
@@ -123,4 +134,11 @@ private void maybeRefreshAdvertisingID(Context context) {
             branch.setGAParamsFetchInProgress(branch.getDeviceInfo().getSystemObserver().prefetchAdsParams(context, branch));
         }
     }
+
+    // default is true
+    boolean isCurrentActivityLaunchedFromStack() {
+        Branch branch = Branch.getInstance();
+        if (branch == null || branch.getCurrentActivity() == null) return true;
+        return activitiesOnStack.contains(branch.getCurrentActivity().getLocalClassName());
```

Comment:
The semantics here are unclear to me.   If the current activity == null, why is it considered "launched from stack".

Reply:
In reality, this should never happen but you are right, semantically, it should return false in that case. 👍